### DISCUSSION TITLE
Unable to get IDfmCustomizedRendererPartProvider example to work

### DISCUSSION THREAD

**Discussion by Stoffelche** [dotnet/docfx#8356] [Q&A]
I am trying to get the example described in "Customize DFM Engine" to work. But the plugin is not being loaded at all.
I don't know what to do. I generated a class library project named TestPlugin targetin core 6.0 and added the 2 classes as described. Then I copied the output files (trying including the Microsoft.DocAsCode files and not including them) to a folder x/plugins below the folder where my docfx.json is. 

Then I put 
`	   "template":
      [
        "default",
		"x"
      ],
` in doxfx.json. 
But nothing happens.
Markdown engine is dfm, so this is not the cause.
Some help would be highly appreciated, as I am stuck.
I am starting docfx from debugger, but was unable to find a location, where to set a breakpoint, in order to find out why it does not work.

**Comment by paulushub** [comment]
dfm engine is depreciated.

**Reply by Stoffelche** [reply]
I am using 2.6. Why should it be deprecated?

**Reply by Stoffelche** [reply]
And else, how can I proceed to process a custom code block in a md-file?

**Reply by paulushub** [reply]
> I am using 2.6. Why should it be deprecated?

`dfm` was a Microsoft implementation of the Markdown parser for the DocFX and is not as complete as the `markdig`.
Some features in DocFX requires the `markdig`, so there is no need to keep `dfm`.

> And else, how can I proceed to process a custom code block in a md-file?

Look into the [markdig](https://github.com/xoofx/markdig), it already provides a lot of extensions to many known cases.
If you still need a custom processing, then consider writing a plugin/extension to the `markdig`.

**Reply by Stoffelche** [reply]
Thank you for your reply, I really appreciate your support. 
I have written a tool, that generates plantuml code for inheritance diagrams for c# types from roslyn compilation. 
Now I wanted to use and enhance the  https://github.com/KevReed/DocFx.Plugins.PlantUml plugin to also support plantuml include files and replace them with svg generated from the plantuml code.
But the plugin would not work with 2.60. So I resorted to the basic example, on which this tool is based, but could not get it running either.
So now I am stuck here.
If I don't find a solution integrated in docfx, I am going  to try using https://github.com/KevReed/PlantUml.Net and some custom preprocessing to replace some custom file link in my md files with svg, hoping that markdig copes with inline svg inside the md files.
In a later step I want to try make the diagram items (types and members) xref enabled, in order to link with api documentation, but I don't know whether xref will be resolved inside svg.
